tests/phpunit/tests/block-bindings/render.php

```diff
@@ -157,6 +157,94 @@ public function test_passing_uses_context_to_source() {
 		);
 	}
 
+	/**
+	 * Tests passing `uses_context` as argument to the source.
+	 *
+	 * @ticket 99999
+	 *
+	 * @covers ::register_block_bindings_source
+	 */
+	public function test_passing_uses_context_to_multiple_sources() {
+		register_block_bindings_source(
+			'test/source-one',
+			array(
+				'label'              => 'Test Source One',
+				'get_value_callback' => function ( $source_args, $block_instance, $attribute_name ) {
+					$value = $block_instance->context['sourceOneContext'];
+					return "Source One value: $value";
+				},
+				'uses_context'       => array( 'commonContext', 'sourceOneContext' ),
+			)
+		);
+
+		register_block_bindings_source(
+			'test/source-two',
+			array(
+				'label'              => 'Test Source Two',
+				'get_value_callback' => function ( $source_args, $block_instance, $attribute_name ) {
+					$value = $block_instance->context['commonContext'];
+					return "Source Two value: $value";
+				},
+				'uses_context'       => array( 'commonContext', 'sourceTwoContext' ),
+			)
+		);
+
+		$block_content                               = <<<HTML
+<!-- wp:paragraph {"metadata":{"bindings":{"content":{"source":"test/source-one", "args": {"key": "test"}}}}} -->
+<p>First source value placeholder</p>
+<!-- /wp:paragraph -->
+<!-- wp:paragraph {"metadata":{"bindings":{"content":{"source":"test/source-two", "args": {"key": "test"}}}}} -->
+<p>Second source value placeholder</p>
+<!-- /wp:paragraph -->
+HTML;
+		[ $first_parsed_block, $second_parsed_block] = array_values(
+			array_filter(
+				parse_blocks( $block_content ),
+				function ( $block ) {
+					return 'core/paragraph' === $block['blockName'];
+				}
+			)
+		);
+
+		$first_block         = new WP_Block(
+			$first_parsed_block,
+			array(
+				'commonContext'    => 'common context value',
+				'sourceOneContext' => 'source one context value',
+			)
+		);
+		$first_block_render  = $first_block->render();
+		$second_block        = new WP_Block(
+			$second_parsed_block,
+			array(
+				'commonContext'    => 'common context value',
+				'sourceTwoContext' => 'source two context value',
+			)
+		);
+		$second_block_render = $second_block->render();
```

Comment:
It's enough to test a single block with the current implementation.

Reply:
You're right that it feels repetitive. I've removed the "multiple sources" test and just rely on the one testing uses_context: [link](https://github.com/SantosGuillamot/wordpress-develop/blob/6f0a931adbfdd7691c1963215a0b32fddb28957a/tests/phpunit/tests/block-bindings/render.php#L124-L158).

 I changed it for another test to ensure that the second source can't access the context defined by the first one.

Let me know what you think.